tableau_utilities/scripts/connection.py

```diff
@@ -1,83 +1,68 @@
 import os
-import sys
 from pprint import pprint
 
 
-import tableau_utilities as tu
+from tableau_utilities.tableau_file.tableau_file import Datasource
 from tableau_utilities.tableau_server.tableau_server import TableauServer
-from tableau_utilities.scripts.server_operate import get_object_list, fill_in_id_name_project
 
 
-def connection_settings(args, debugging_logs, settings):
+def connection_settings(args, debugging_logs, settings=None):
     """ Creates a dictionary with named connection information
 
-    Returns:
-        connection_settings: Dictionary with the connection information
-    """
+    Args:
+        args: An argparse args object
+        debugging_logs (bool): True to print logging messages
+        settings (dict): A dict of settings from a local file
 
-    conn = args.conn
+    Returns: Dictionary with the connection information
+    """
+    if not settings:
+        settings = {'embed_connection': dict()}
 
     if debugging_logs:
         print('Building the connection settings')
 
-    conn_user = args.conn_user
-    conn_pw = args.conn_pw
-    conn_type = args.conn_type
-    conn_db = args.conn_db
-    conn_schema = args.conn_schema
-    conn_host = args.conn_host
-    conn_role = args.conn_role
-    conn_warehouse = args.conn_warehouse
-
-    if conn == 'settings_yaml':
-        if debugging_logs:
-            print("Building connection settings from yaml file")
-        conn_user = settings['embed_connection']['username']
-        conn_pw = settings['embed_connection']['password']
-        conn_type = settings['embed_connection']['class_name']
-        conn_db = settings['embed_connection']['dbname']
-        conn_schema = settings['embed_connection']['schema']
-        conn_host = settings['embed_connection']['server']
-        conn_role = settings['embed_connection']['service']
-        conn_warehouse = settings['embed_connection']['warehouse']
-    elif conn == 'os_env':
-        if debugging_logs:
-            print("Building connection settings from environment variables")
-        conn_user = os.getenv("CONN_USERNAME")
-        conn_pw = os.getenv("CONN_PASSWORD")
-        conn_type = os.getenv("CONN_TYPE")
-        conn_db = os.getenv("CONN_DB")
-        conn_schema = os.getenv("CONN_SCHEMA")
-        conn_host = os.getenv("CONN_HOST")
-        conn_role = os.getenv("CONN_ROLE")
-        conn_warehouse = os.getenv("CONN_WAREHOUSE")
-
-
-    connection_settings = {'conn_user': conn_user,
-                           'conn_pw': conn_pw,
-                           'conn_type': conn_type,
-                           'conn_db': conn_db,
-                           'conn_schema': conn_schema,
-                           'conn_host': conn_host,
-                           'conn_role': conn_role,
-                           'conn_warehouse': conn_warehouse
-                           }
+    conn_settings = {
```

Comment:
Same comments as in the CLI.  I understand how this is less code but it feels harder to debug without the `if` statement explicitly saying which bucket of credentials to get.